---
title: "Chapter8.Rmd"
author: "Jonathan Pevsner, modified by Keith Hultman"
date: "June 10, 2016"
output: html_document
---
 
# Bioinformatics and Functional Genomics (3rd edition)   

## Computer lab 8.3

#### First obtain the data from the UCSC table browser following the instructions in the text.

Follow the directions in the text. One change from the text, for "group" select "Repeats" instead of "Variations and Repeats". This will give you the repeats found within the beta globin locus of human.

#### Upload the file into RStudio

After downloading the text file to your computer, change the name of the file to `ucsc_chr11_repeats`. Then, use the Upload button on the RStudio Files window to select your file and upload it. 

#### Step 2, Option 1: Next download a file containing chromosome 11 repeats into this directory. Confirm the file is present using dir().

We'll load the file into R (using the read.delim function), we'll call our new table myRepeats (you can assign it almost any name you prefer), specify that the table does have a header, and specify that it is tab-delimited (using sep = "\t"). Get help on a function such as read.delim by entering ?read.delim at the R command prompt.

```{r import from local}
dir() # Check to see if your file is here, if it's not something is wrong.
myRepeats <- read.delim("ucsc_chr11_repeats", header = TRUE, sep = "\t")
```

#### Now we'll look at the file various ways, and change the names of the column headers.

```{r look}
head(myRepeats)
# Change the names of the headers to something easier to read/understand
colnames(myRepeats) <- c("swScore", "start", "end", "strand", "name", "class", "family")
# Look at new header names
head(myRepeats)
# Summary stats for each column
summary(myRepeats)
# Rows by Columns:
dim(myRepeats)
# Show types of variables in each column:
str(myRepeats)
```

#### Plot the repeat classes as a boxplot.

```{r repeat plot}
library(ggplot2)
# create a plot object with your data and parameters for the plot
plot <- ggplot(data = myRepeats, aes(x= class, y= swScore)) +
  geom_boxplot(fill= "pink") +
  ylim(c(0,27500)) +
  ggtitle("Repeat classes in the human beta globin locus") +
  xlab("Class") +
  ylab("SW Score")

# print the plot  
plot  
```

#### Use tapply to apply the functions mean and range over the entire array.

```{r tapply}
tapply(myRepeats$swScore, myRepeats$class, mean)
tapply(myRepeats$swScore, myRepeats$class, range)
```

### Questions

1. Remake the plot for the human kit locus. Use the lookup function of the position text field by searching for 'kit'. Download the same file type as before, then redo the analysis with that locus. 

2. What is the mean SW score for the LINE class of kit and beta globin?

## Computer lab 8.4

In this exercise we will use the R package GenomeGraphs in RStudio to plot the structure of the beta globin gene, and plot the position of this gene on an ideogram of chromosome 11. We will extract information from the R package biomaRt. For more information browse the GenomeGraphs vignette at the bioconductor.org website, as well as a user's guide written by Steffen Durinck and James Bullard.

#### Load GenomeGraphs

```{r prelim, cache=TRUE}
library(GenomeGraphs)
```

#### Use biomaRt to obtain a gene structure that we will plot.

```{r mart, cache=TRUE}
mart <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
gene <- makeGene(id = "ENSG00000244734", type="ensembl_gene_id", biomart = mart)
gdPlot(gene) # save the output as Rplot1 (a .png file)

```

#### Next plot transcripts associated with this gene. Save the output as Rplot2 (a .png file).

```{r plot2, cache=TRUE}
transcript <- makeTranscript(id = "ENSG00000244734", type="ensembl_gene_id", biomart = mart)
gdPlot(list(gene, transcript))
```

#### Define the genomic region. The resulting plot has brown boxes for [exons] and genomic coordinates. Save it as Rplot3.
Here we need to enter the coordinates around the gene. I found the coordinates on ensembl search for the gene id.
```{r plot3, cache=TRUE}
minusStrand <- makeGeneRegion(chromosome = 11, start = 5225464, end = 5229395, strand = "-", biomart = mart)
genomeAxis <- makeGenomeAxis(add53 = TRUE) # Add53 shows 5' and 3' ends
gdPlot(list(genomeAxis, minusStrand)) 
```

#### Add an ideogram of chromosome 11 to the plot.
Here we will add context to the gene display. We will make a full chromosome display, top strand transcripts, the sequence location track, our gene, and the rest of the bottom strand transcripts. The full chromosome display will show where we are looking at the chromosome by a red box. We are looking at a very small region so the box will look like a line.  

```{r plot23, echo=FALSE}
topStrand <- makeGeneRegion( chromosome = 11, start = 5200000, end = 5250000, strand = "+", biomart = mart)
fullminusStrand <- makeGeneRegion( chromosome = 11, start = 5200000, end = 5250000, strand = "-", biomart = mart)
ideogram <- makeIdeogram(chromosome = 11)
genomeAxis <- makeGenomeAxis(add53=TRUE, add35=TRUE)
gdPlot(list(ideogram, topStrand, genomeAxis, minusStrand, fullminusStrand)) 
# save as Rplot4.png
```


## Computer lab 8.11

The dinucleotide CG is often referred to as CpG (p denotes the phosphate linkage between the two residues). In the human genome CpG dinucleotides occur at a very low frequency (about five-fold less than other dinucleotides). What are the frequencies of all dinucleotides at the human beta globin (HBB) locus on chromosome 11? 

To answer this, we will obtain this DNA sequence in the FASTA format, import it into the R package SeqinR, and use the count function. This problem is described in an online book by Avril Coghlan available at http://a-little-book-of-r-for-bioinformatics.readthedocs.org/en/latest/src/chapter1.html  See also the SeqinR documentation from http://SeqinR.r-forge.r-project.org/.

Select a region of 60,000 base pairs on chr11:5,240,001-5,300,000, encompassing HBB and other globin genes. You can access the DNA via the Table Browser (select output format > sequence) or from the Genome Browser (view > DNA). You can also view SeqinR documentation for instructions on fetching sequences from NCBI in R. 


#### Load the SeqinR library.

```{r install, cache=TRUE}
library("seqinr")
```

#### Import the sequence data. 

We read the FASTA formatted file into an R object called globinDNA.
See text for how to obtain the sequence. 
```{r dir, cache=TRUE}
#dir() # Looking at at the directory we confirm the sequence file is present
globinDNA <- read.fasta(file = "chr11_60kb")
globinseq <- globinDNA[[1]]
```

#### Confirm the length of this sequence is 60 kb.

```{r length, cache=TRUE}
length(globinseq)  
```

#### The count function reports the frequency of each nucleotide.

```{r count}
count(globinseq,1) 
```

#### We specify we want to know the frequency of all dinucleotides.

```{r dinucl}
count(globinseq,2)
```

#### Note that the frequency of CpG dinucleotides is indeed substantially lower than that of all other dinucleotides. We can also create a table object with these results (called mydinucleotides) and view a particular result. We can also plot the results.

```{r CpG plot, cache=TRUE}
mydinucleotides <- count(globinseq,2)
mydinucleotides[["cg"]]
plot(mydinucleotides)
```

#### Questions
1. Graph the relative frequencies of all dinucleotides instead of the absolute counts. Hint: look at the documentation for the sequinr:count function to find out how to return frequencies: "?count"


## Show the session information
You should always have this at the end of your experiments for reproducible research purposes!
```{r}
sessionInfo()
```
