---
title: "Problem 8.4"
author: "Jonathan Pevsner, edited by Keith Hultman"
date: "June 10, 2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Bioinformatics and Functional Genomics  

#### In this exercise we will use the R package GenomeGraphs in RStudio to plot the structure of the beta globin gene, and plot the position of this gene on an ideogram of chromosome 11. We will extract information from the R package biomaRt. For more information browse the GenomeGraphs vignette at the bioconductor.org website, as well as a user's guide written by Steffen Durinck and James Bullard.

#### Load GenomeGraphs

```{r prelim, cache=TRUE}
library(GenomeGraphs)
```

#### Use biomaRt to obtain a gene structure that we will plot.

```{r mart, cache=TRUE}
mart <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
gene <- makeGene(id = "ENSG00000244734", type="ensembl_gene_id", biomart = mart)
gdPlot(gene) # save the output as Rplot1 (a .png file)

```

#### Next plot transcripts associated with this gene. Save the output as Rplot2 (a .png file).

```{r plot2, cache=TRUE}
transcript <- makeTranscript(id = "ENSG00000244734", type="ensembl_gene_id", biomart = mart)
gdPlot(list(gene, transcript))
```

#### Define the genomic region. The resulting plot has brown boxes for [exons] and genomic coordinates. Save it as Rplot3.
Here we need to enter the coordinates around the gene. I found the coordinates on ensembl search for the gene id.
```{r plot3, cache=TRUE}
minusStrand <- makeGeneRegion(chromosome = 11, start = 5225464, end = 5229395, strand = "-", biomart = mart)
genomeAxis <- makeGenomeAxis(add53 = TRUE) # Add53 shows 5' and 3' ends
gdPlot(list(genomeAxis, minusStrand)) 
```

#### Add an ideogram of chromosome 11 to the plot.
Here we will add context to the gene display. We will make a full chromosome display, top strand transcripts, the sequence location track, our gene, and the rest of the bottom strand transcripts. The full chromosome display will show where we are looking at the chromosome by a red box. We are looking at a very small region so the box will look like a line.  
```{r plot23}
topStrand <- makeGeneRegion( chromosome = 11, start = 5200000, end = 5250000, strand = "+", biomart = mart)
fullminusStrand <- makeGeneRegion( chromosome = 11, start = 5200000, end = 5250000, strand = "-", biomart = mart)
ideogram <- makeIdeogram(chromosome = 11)
genomeAxis <- makeGenomeAxis(add53=TRUE, add35=TRUE)
gdPlot(list(ideogram, topStrand, genomeAxis, minusStrand, fullminusStrand)) 
# save as Rplot4.png
```


#### Show the session information
```{r}
sessionInfo()
```

